containers/src/Kokkos_DynRankView.hpp

```diff
@@ -614,6 +614,9 @@ class DynRankView : private View<DataType*******, Properties...> {
     } else
 #endif
       return view_type::operator()(i0, 0, 0, 0, 0, 0, 0);
+#ifdef KOKKOS_COMPILER_INTEL
```

Comment:
hmm shouldn't we check if we can also put this for different compilers?

Reply:
That's fair.  I can find both `#ifdef KOKKOS_COMPILER_INTEL` or
```C++
  #if defined KOKKOS_COMPILER_INTEL ||                                  \
      (defined(KOKKOS_COMPILER_NVCC) && KOKKOS_COMPILER_NVCC >= 1130 && \
       !defined(KOKKOS_COMPILER_MSVC))
```
guarding `__builtin_unreachable();` throughout the code base.

@ndellingwood please confirm that all nightlies with NVCC are passing with this PR

Reply:
The nvcc nightlies have been passing, but I can add the change you suggested to be safe

Reply:
I updated the guards here [9540583](https://github.com/kokkos/kokkos/pull/7614/commits/95405836838efbdae183756830c83d543218f336)